resources/wikia/modules/viewportObserver.js

```diff
@@ -15,6 +16,34 @@ define('wikia.viewportObserver', [
 		}
 	}
 
+	function sameViewport(element, otherElements) {
+		element = doc.getElementById(element);
+
+		var windowHeight = win.innerHeight || doc.documentElement.clientHeight || doc.body.clientHeight,
+			elementOffset = domCalculator.getTopOffset(element),
+			elementHeight = element.offsetHeight,
+			found = false;
+
+		otherElements.forEach(function (other) {
+			other = doc.getElementById(other);
+
+			var otherOffset = domCalculator.getTopOffset(other),
+				otherHeight = other.offsetHeight;
+
+			if (elementOffset < otherOffset) {
```

Comment:
I think you can easily reduce the number of ifs by extracting ```otherOffset - elementOffset - elementHeigh```
and ``` elementOffset - otherOffset - otherHeight ``` to variable (even into to ternary operator, but then it might be a little bit messy).

Reply:
I made some cleanup :)